### PR TITLE
require search before displaying results

### PR DESCRIPTION
**Pull request by kielni**
- starts.geojson endpoint returns 400 if no lat/lng params
- find page gets carpools (starts.geojson) only when one of
  - lat/lng in URL
  - user geolocation coordinates available
  - user query geocoded
- if no lat/lng from user (via geolocation or text query), show no results

for https://github.com/RagtagOpen/nomad/issues/249

### CODE REVIEW

**Comment by kielni** [comment]
@jillh510 this still returns all the carpools; should I limit results by distance, count, or both?

**Comment by kielni** [comment]
from Slack

> we're agreed on limiting returned values via the API to carpools within a 50 mile radius. I'd also like to hear any objections to the idea that logged-out users see at most 3 carpools

I'll update this PR with those rules today/tomorrow.

File: app/carpool/views.py [hunk 1]
```diff
@@ -77,29 +77,19 @@ def start_geojson():
         pools = pools.filter(Carpool.leave_time >= datetime.datetime.utcnow())
 
     try:
-        near_lat = request.args.get('near.lat', type=float)
-        near_lon = request.args.get('near.lon', type=float)
+        near_lat = request.args.get('near.lat', None, type=float)
```
Comment (iandees) [inline app/carpool/views.py]: `None` is the default return value already. You shouldn't have to specify it here.

**Comment by dryan** [comment]
@kielni this completely addresses #249 right?

**Pull request merged**